lib/matplotlib/path.py

```diff
@@ -322,18 +322,17 @@ def make_compound_path(cls, *args):
         Make a compound path from a list of `Path` objects. Blindly removes
         all `Path.STOP` control points.
         """
-        # Handle an empty list in args (i.e. no args).
         if not args:
             return Path(np.empty([0, 2], dtype=np.float32))
         vertices = np.concatenate([x.vertices for x in args])
         codes = np.empty(len(vertices), dtype=cls.code_type)
         i = 0
         for path in args:
             if path.codes is None:
-                codes[i] = cls.MOVETO
-                codes[i + 1:i + len(path.vertices)] = cls.LINETO
+                codes[i : i+1] = cls.MOVETO
```

Comment:
Note that this is the relevant change in this file: it makes the function work even if codes has length zero.